cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -206,6 +206,7 @@ def execute(service_name = nil, name: loader.stack_name.stack, parent_name: nil,
           result['stack']         = raw_yaml['stack']
           result['version']       = loader.stack_name.version || '0.0.1'
           result['name']          = name
+          result['labels']        = raw_yaml['labels'] || []
```

Comment:
This needs to be:

`fully_interpolated_yaml['labels'] || []` 

The `raw_yaml` is from before any variable/liquid interpolation has taken place. With `fully_interpolated_yaml` it's possible to insert variable values into labels.

For example:
```yaml
variables:
  customer:
    type: string
    from:
      prompt: Enter customer name
labels:
  - customer=${customer}
```

Reply:
@kke thanks for the catch - that's a good one!